cmd/snap/cmd_model.go

```diff
@@ -125,19 +144,77 @@ func (x *cmdModel) Execute(args []string) error {
 
 	w := tabWriter()
 
+	if x.Serial && client.IsAssertionNotFoundError(serialErr) {
+		// for serial assertion, the primary keys are output (model and
+		// brand-id), but if we didn't find the serial assertion then we still
+		// output the brand-id and model from the model assertion, but also
+		// return a devNotReady error
+		fmt.Fprintf(w, "brand-id:\t%s\n", modelAssertion.HeaderString("brand-id"))
+		fmt.Fprintf(w, "model:\t%s\n", modelAssertion.HeaderString("model"))
+
+		return errors.New(devNotReadyMessage)
+	}
+
+	// the rest of this function is the main flow for outputting either the
+	// model or serial assertion in normal or verbose mode
+
 	// output the primary keys first in their canonical order
-	for _, headerName := range assertion.Type().PrimaryKey {
-		if headerName == "series" {
+	for _, headerName := range mainAssertion.Type().PrimaryKey {
+		switch headerName {
+		case "series":
 			// series can be obtained from "snap version"
 			// and given it is stuck at 16 is mainly confusing
-			continue
+		case "model":
+			if x.Serial {
+				fmt.Fprintf(w, "model:\t%s\n", mainAssertion.HeaderString("model"))
+				continue
+			}
+
+			// for the model command (not --serial) we want to conditionally
+			// use the display-name for the model if it exists (with the
+			// normal model header after the display-name)
+			modelHeader := modelAssertion.HeaderString("model")
+			if displayName := modelAssertion.HeaderString("display-name"); displayName != "" {
+				modelHeader = fmt.Sprintf("%s (%s)", displayName, modelHeader)
+			}
+			fmt.Fprintf(w, "model:\t%s\n", modelHeader)
+		case "brand-id":
+			brandID := mainAssertion.HeaderString("brand-id")
+			if x.Serial {
+				fmt.Fprintf(w, "brand-id:\t%s\n", brandID)
+				continue
+			}
+
+			// for the model command (not --serial) we want to output the
+			// "brand" instead of the "brand-id"
+			storeAccount, err := x.client.StoreAccount(brandID)
+			if err != nil {
+				return err
+			}
+			// use the longPublisher helper to format the brand store account
+			// like we do in `snap info`
+			fmt.Fprintf(w, "brand:\t%s\n", longPublisher(x.getEscapes(), storeAccount))
+		default:
+			fmt.Fprintf(w, "%s:\t%s\n", headerName, mainAssertion.HeaderString(headerName))
+		}
+	}
+
+	if !x.Serial {
+		// for the model command we also need to always add the serial, even if
+		// we don't have a serial assertion yet
+		var serial string
+		if client.IsAssertionNotFoundError(serialErr) {
+			// didn't get a serial assertion yet
+			serial = "- (device not registered yet)"
+		} else {
+			serial = serialAssertion.HeaderString("serial")
 		}
-		fmt.Fprintf(w, "%s:\t%s\n", headerName, assertion.HeaderString(headerName))
+		fmt.Fprintf(w, "serial:\t%s\n", serial)
```

Comment:
no colon if this the "snap model (no options)" case

Reply:
changed to see how well it works